Userland/Applications/ImageViewer/ViewWidget.cpp

```diff
@@ -60,49 +61,58 @@ void ViewWidget::rotate(Gfx::RotationDirection rotation_direction)
     resize_window();
 }
 
-void ViewWidget::navigate(Directions direction)
+bool ViewWidget::is_next_available() const
 {
-    if (m_path == nullptr)
-        return;
+    if (m_current_index.has_value())
+        return m_current_index.value() + 1 < m_files_in_same_dir.size();
+    return false;
+}
+
+bool ViewWidget::is_previous_available() const
+{
+    if (m_current_index.has_value())
+        return m_current_index.value() > 0;
+    return false;
+}
 
-    auto current_dir = LexicalPath(m_path).parent().string();
+Vector<String> ViewWidget::load_files_from_directory(const String& path) const {
+    Vector<String> files_in_directory;
 
-    if (m_files_in_same_dir.is_empty()) {
-        Core::DirIterator iterator(current_dir, Core::DirIterator::Flags::SkipDots);
-        while (iterator.has_next()) {
-            String file = iterator.next_full_path();
-            if (!Gfx::Bitmap::is_path_a_supported_image_format(file))
-                continue;
-            m_files_in_same_dir.append(file);
-        }
+    auto current_dir = LexicalPath(path).parent().string();
+    Core::DirIterator iterator(current_dir, Core::DirIterator::Flags::SkipDots);
+    while (iterator.has_next()) {
+        String file = iterator.next_full_path();
+        if (!Gfx::Bitmap::is_path_a_supported_image_format(file))
+            continue;
+        files_in_directory.append(file);
     }
+    return files_in_directory;
+}
 
-    auto current_index = m_files_in_same_dir.find_first_index(m_path);
-    if (!current_index.has_value()) {
+void ViewWidget::initialize_data(const String& path){
```

Comment:
Just one thing that seems odd, this function's name feels way too generic, could we come up with a more descriptive name for it?

Reply:
what about `initialize_path_data`?

Reply:
Hmmm...`initialize` just sounds weird I guess, can we get away with `set_path()`?

Reply:
`set_path_related_data`?

Reply:
I don't think its users really need to know what it does other than setting the path, let's just call it `set_path()`.

Reply:
Make sense.